lib/galaxy/datatypes/binary.py

```diff
@@ -4297,9 +4297,9 @@ def __init__(self, **kwd):
 
     def sniff(self, filename: str) -> bool:
         try:
-            npz = np.load(filename)
-            if isinstance(npz, np.lib.npyio.NpzFile):
-                return True
+            with np.load(filename) as npz:
+                if isinstance(npz, np.lib.npyio.NpzFile) and any(f.filename.endswith(".npy") for f in npz.zip.filelist):
+                    return True
```

Comment:
Nice trick with `npz.zip` :) 

Should be `all(f.filename.endswith(".npy") ...` or? 


Or maybe faster? 

```suggestion
                if isinstance(npz, np.lib.npyio.NpzFile):
                    for f in npz.zip.filelist:
                        if not f.filename.endswith(".npy"):
                            return False
                    return True
```

Reply:
It's valid for non npy files to be included:

> It assumes that files in the archive have a .npy extension, other files are ignored.

Reply:
This is not faster, `any` does that already internally: https://docs.python.org/3/library/functions.html#any

Reply:
Got it.